crates/ra_ide_api/src/goto_type_definition.rs

```diff
@@ -30,9 +30,12 @@ pub(crate) fn goto_type_definition(
         return None;
     };
 
-    let (adt_def, _) = ty.as_adt()?;
-    let nav = NavigationTarget::from_adt_def(db, adt_def);
+    let adt_def = ty.autoderef(db).find_map(|ty| match ty.as_adt() {
+        Some((adt_def, _)) => Some(adt_def),
+        None => None,
+    })?;
```

Comment:
I would have written it as `let (adt_def, _subst) = ty.autoderef(db).find_map(|ty| ty.as_adt())?`

It seems a tiny bit  simpler that way, and given `_names` to ignored stuff is just a touch more readable

Reply:
r=me with or without this change

Reply:
That is much cleaner. I'm running into the borrow checker borrowing `ty` since the `substs` is returned by reference.

Reply:
@matklad This appeases the borrow checker: `let adt_def = ty.autoderef(db).find_map(|ty| ty.as_adt().map(|adt| adt.0))?;` Is that reasonable?